src/scf/self_consistent_field.jl

```diff
@@ -6,74 +6,44 @@ include("scf_callbacks.jl")
     verbose = false
 end
 
-function default_n_bands(model)
-    n_spin = model.n_spin_components
-    min_n_bands = div(model.n_electrons, n_spin * filled_occupation(model), RoundUp)
-    n_extra = model.temperature == 0 ? 0 : max(4, ceil(Int, 0.2 * n_spin * min_n_bands))
-    min_n_bands + n_extra
-end
-default_occupation_threshold() = 1e-6
-
 """
-Obtain new density ρ by diagonalizing `ham`.
-"""
-function next_density(ham::Hamiltonian;
-                      n_bands=default_n_bands(ham.basis.model),
-                      ψ=nothing, n_ep_extra=3,
-                      eigensolver=lobpcg_hyper,
-                      occupation_threshold,
-                      kwargs...)
-    if ψ !== nothing
-        @assert length(ψ) == length(ham.basis.kpoints)
-        for ik in 1:length(ham.basis.kpoints)
-            @assert size(ψ[ik], 2) == n_bands + n_ep_extra
-        end
-    end
-
-    # Diagonalize
-    eigres = diagonalize_all_kblocks(eigensolver, ham, n_bands + n_ep_extra; ψguess=ψ,
-                                     n_conv_check=n_bands, kwargs...)
-    eigres.converged || (@warn "Eigensolver not converged" iterations=eigres.iterations)
-
-    # Update density from new ψ
-    occupation, εF = compute_occupation(ham.basis, eigres.λ; occupation_threshold)
-    ρout = compute_density(ham.basis, eigres.X, occupation)
-
-    (ψ=eigres.X, eigenvalues=eigres.λ, occupation=occupation, εF=εF,
-     ρout=ρout, diagonalization=eigres)
-end
+Solve the Kohn-Sham equations with a SCF algorithm, starting at `ρ`.
 
-
-"""
-Solve the Kohn-Sham equations with a SCF algorithm, starting at ρ.
+- `bands`: By default DFTK uses an adaptive algorithm to determine the number of bands
+  to compute. If `bands` is an integer, it specifies the minimal number of bands to be
```

Comment:
that seems potentially confusing. Isn't it better to do the fixed thing when bands is an integer?

Reply:
I'm not sure. Most people will want the adaptive thing.

I view the FixedBands more as a debug / preformance-tracking option, so more expert. On the other hand I'm pretty confident that manual adjustments of the bands should now be needed a lot less regularly (i.e. basically only if you want to do something special), so perhaps we should drop the integer thing alltogether?